File: src/DataScience/MATLAB/singleton.m
function singleton()
    first = registry_instance();
    second = registry_instance();
    first.increment();

    fprintf('same=%s\n', lower(string(first == second)));
    fprintf('count=%d\n', second.count());
end

function value = registry_instance()
    persistent shared
    if isempty(shared) || ~isvalid(shared)
        shared = RegistryState();
    end
    value = shared;
end

File: src/DataScience/R/singleton.R
registry <- new.env(parent = emptyenv())
assign("count", 0L, envir = registry)

registry_instance <- function() registry

first <- registry_instance()
second <- registry_instance()
assign("count", get("count", envir = first) + 1L, envir = first)

cat(sprintf("same=%s\n", tolower(as.character(identical(first, second)))))
cat(sprintf("count=%d\n", get("count", envir = second)))
